dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1437,14 +1444,10 @@ private ExecutionStatus processReadyPause() {
     private ExecutionStatus getProcessInstanceState(ProcessInstance instance) {
         ExecutionStatus state = instance.getState();
 
-        if (activeTaskProcessorMaps.size() > 0 || hasRetryTaskInStandBy()) {
+        if ((activeTaskProcessorMaps.size() > 0 || hasRetryTaskInStandBy()) && state != ExecutionStatus.READY_STOP) {
```

Comment:
Is it better to put the process instance state check before?

Reply:
This has been check in readyToSubmitTaskQueue.clear(). I'll remove this logic.